### PR TITLE
Feature/protect env

### PR DESCRIPTION
**Pull request by adrien-cotte**
Hi,

Here is my new PR for issue https://github.com/cea-hpc/modules/issues/429.
Not sure about test suite.

Best,
Adrien

### CODE REVIEW

**Comment by adrien-cotte** [comment]
Hi @xdelaruelle,

Could you help me to understand what's wrong with my tests, please?

Best,
Adrien

File: testsuite/modules.70-maint/410-protected_envvars.exp [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+set modfilere $mpre/$mod
+
+#
+#  Tests
+#
+
+set ans [list]
+lappend ans [list set _LMFILES_ $modfile]
+lappend ans [list set LOADEDMODULES $mod]
+lappend ans [list ERR]
```
Comment (xdelaruelle) [inline testsuite/modules.70-maint/410-protected_envvars.exp:39]: The introduced protected variable mechanism emits a warning message when trying to modify a variable set as protected. This mechanism does not lead to an error, so the tests do not output an error shell code. If you remove this `lappend ans [list ERR]` line, the tests should pass.

Comment (adrien-cotte) [reply testsuite/modules.70-maint/410-protected_envvars.exp]: Damn it, so obvious...
Many thanks, again.

**Comment by xdelaruelle** [comment]
Many thanks for your contribution @adrien-cotte

**Pull request merged**